test/Microsoft.ML.Functional.Tests/Training.cs

```diff
@@ -0,0 +1,534 @@
+            var featurization = featurizationPipeline.Fit(data);
+            var featurizedData = featurization.Transform(data);
+
+            // Fit the first trainer.
+            var firstModel = trainer.Fit(featurizedData);
+            var firstModelWeights = firstModel.Model.Weights;
+
+            // Fist the first trainer again.
+            var firstModelPrime = trainer.Fit(featurizedData);
+            var firstModelWeightsPrime = firstModel.Model.Weights;
+
+            // Fit the second trainer.
+            var secondModel = trainer.Fit(featurizedData, firstModel.Model);
+            var secondModelWeights = secondModel.Model.Weights;
+
+            // Validate that continued training occurred.
+            // Training from the same initial condition, same seed should create the same model.
+            Common.AssertEqual(firstModelWeights.ToArray(), firstModelWeightsPrime.ToArray());
+            // Continued training should create a different model.
+            Common.AssertNotEqual(firstModelWeights.ToArray(), secondModelWeights.ToArray());
+        }
+
+        /// <summary>
+        /// Training: Meta-compononts function as expected. For OVA (one-versus-all), a user will be able to specify only
+        /// binary classifier trainers. If they specify a different model class there should be a compile error.
+        /// </summary>
+        [Fact]
+        public void MetacomponentsFunctionAsExpectedOva()
+        {
+            var mlContext = new MLContext(seed: 1);
+
+            var data = mlContext.Data.LoadFromTextFile<Iris>(GetDataPath(TestDatasets.iris.trainFilename),
+                            hasHeader: TestDatasets.iris.fileHasHeader,
+                            separatorChar: TestDatasets.iris.fileSeparator);
+
+            // Create a model training an OVA trainer with a binary classifier.
+            var anomalyDetectionTrainer = mlContext.AnomalyDetection.Trainers.AnalyzeRandomizedPrincipalComponents();
+            var anomalyDetectionPipeline = mlContext.Transforms.Concatenate("Features", Iris.Features)
+                .AppendCacheCheckpoint(mlContext)
+                .Append(mlContext.Transforms.Conversion.MapValueToKey("Label"))
+                .Append(mlContext.MulticlassClassification.Trainers.OneVersusAll(anomalyDetectionTrainer))
+                .Append(mlContext.Transforms.Conversion.MapKeyToValue("PredictedLabel"));
+
+            // Fit the binary classification pipeline.
+            Assert.Throws<InvalidOperationException>(() => anomalyDetectionPipeline.Fit(data));
+
+            // Create a model training an OVA trainer with a binary classifier.
+            var binaryclassificationTrainer = mlContext.BinaryClassification.Trainers.LogisticRegression(
+                new LogisticRegression.Options { NumberOfIterations = 10, NumberOfThreads = 1, });
+            var binaryClassificationPipeline = mlContext.Transforms.Concatenate("Features", Iris.Features)
```

Comment:
>binaryClassificationPipeline [](start = 16, length = 28)

multiclassPipeline?

Reply:
Well, it's supposed be the OVA pipeline made up of binary classifiers.

---
In reply to: [264537263](https://github.com/dotnet/machinelearning/pull/2921#discussion_r264537263) [](ancestors = 264537263)